default/python/AI/ProductionAI.py

```diff
@@ -1082,6 +1087,10 @@ def generate_production_orders():
                 planet = universe.getPlanet(pid)
                 if not planet:
                     continue
+                if not fo.isProducibleBuilding(building_name, pid):
+                    continue
```

Comment:
I think silent failure here will make debugging hard, so please add some logging statement before the continue statement (also below).

Reply:
The idea here isn't that it's a failure, but that it's being used to search for valid places to produce the building. Prior to this change, it just attempts to issue the enqueue order at a bunch of places until one succeeds.